pkg/query-service/app/queryBuilder/query_builder.go

```diff
@@ -147,11 +147,25 @@ func (qb *QueryBuilder) PrepareQueries(params *v3.QueryRangeParamsV3, args ...in
 					if len(args) > 0 {
 						keys = args[0].(map[string]v3.AttributeKey)
 					}
-					queryString, err := qb.options.BuildTraceQuery(params.Start, params.End, compositeQuery.QueryType, compositeQuery.PanelType, query, keys)
-					if err != nil {
-						return nil, err
+					// for ts or table query with group by and limit form two queries
```

Comment:
Do you use two queries for table type?

Reply:
Nope, we are not.
We are just using single query now, planned to use 2 queries earlier. Let me update.